lib/interactor/sidekiq.rb

```diff
@@ -24,6 +24,15 @@ module SidekiqWorker
     class Worker
       include ::Sidekiq::Worker
 
+      sidekiq_retries_exhausted do |job, e|
+        return if jobs['args'].blank? || job['args'].first['interactor_class'].blank?
+
+        interactor_class = job['args'].first['interactor_class'].constantize
+        if interactor_class.respond_to?(:handle_sidekiq_retries_exhausted)
+          interactor_class.handle_sidekiq_retries_exhausted(job, e)
+        end
+      end
```

Comment:
If the following block `#perform` raises en exception, will this block that from getting to Bugsnag unless it's retries are exhausted? Right now we're not using `#handle_sidekiq_exception`, we're just allowing the error to be raised and allowing the system to deal with it, i.e. send it to Bugsnag.

Reply:
`#handle_sidekiq_exception` must be defined in the Interactor. If not it will just `raise` which preserves the existing behavior. That is why I had to add this special handling to our `UpsertRecord` interactor see: https://github.com/instrumentl/instrumentl/pull/7105/files#diff-4c4bb0f01586f51bd37106df8c009e4532206379f09aaf146dd8019e43ca0546R41-R49

Sidekiq still needs the error to be raised to trigger its normal retry logic, but we need to opt out of alerting Bugsnag which happens somewhere up the chain.

Reply:
Can we mimic the behaviour for `#perform`? Then we don't need to add the same handling for every `::Interactor`, and instead it'll log it to Bugsnag but only if it's the final execution.

```suggestion
      sidekiq_retries_exhausted do |job, e|
        return if jobs['args'].blank? || job['args'].first['interactor_class'].blank?

        interactor_class = job['args'].first['interactor_class'].constantize
        if interactor_class.respond_to?(:handle_sidekiq_retries_exhausted)
          interactor_class.handle_sidekiq_retries_exhausted(job, e)
        else
          raise e
        end
      end
```

Reply:
Ahh got it, yes, that makes sense